--- origsrc/audacious-2.1/src/libaudtag/Makefile	2009-07-06 17:39:12.000000000 -0500
+++ src/audacious-2.1/src/libaudtag/Makefile	2009-11-16 21:10:14.096696400 -0600
@@ -21,4 +21,4 @@ CPPFLAGS = ${LIB_CPPFLAGS} ${CFLAGS} -D_
 CFLAGS += ${LIB_CFLAGS}		\
 	  ${GLIB_CFLAGS}	\
 	  ${MOWGLI_CFLAGS}
-LIBS += ${GLIB_LIBS}	
+LIBS += -L../libaudcore -laudcore ${GLIB_LIBS} ${MOWGLI_LIBS}
--- origsrc/audacious-2.1/src/libid3tag/Makefile	2009-07-06 17:39:12.000000000 -0500
+++ src/audacious-2.1/src/libid3tag/Makefile	2009-11-16 21:10:14.096696400 -0600
@@ -31,4 +31,4 @@ CPPFLAGS += ${LIB_CPPFLAGS} -I.. ${GLIB_
 	${MOWGLI_CFLAGS} ${LIBMCS_CFLAGS}
 
 CFLAGS += ${LIB_CFLAGS}
-LIBS += -lz
+LIBS += -L../libaudcore -laudcore ${GLIB_LIBS} -lz
--- origsrc/audacious-2.1/src/libid3tag/file.c	2009-07-06 17:39:12.000000000 -0500
+++ src/audacious-2.1/src/libid3tag/file.c	2009-11-16 21:10:14.096696400 -0600
@@ -46,7 +46,16 @@
 //#undef AUDACIOUS
 
 #ifdef AUDACIOUS
-  #include <audacious/plugin.h>
+  #include <libaudcore/vfs.h>
+  #define aud_vfs_fopen vfs_fopen
+  #define aud_vfs_fclose vfs_fclose
+  #define aud_vfs_dup vfs_dup
+  #define aud_vfs_fseek vfs_fseek
+  #define aud_vfs_ftell vfs_ftell
+  #define aud_vfs_rewind vfs_rewind
+  #define aud_vfs_fread vfs_fread
+  #define aud_vfs_fwrite vfs_fwrite
+  #define aud_vfs_truncate vfs_truncate
 #else
   #define VFSFile FILE
   #define aud_vfs_fopen fopen
